### ISSUE TITLE
UnnecessaryParentReference and & + &

### ISSUE THREAD

**Issue by kaelig** [brigade/scss-lint#217]
Since 

``` scss
.element {
  + & { … }
}
```

Compiles to:

``` scss
+ .element {
  …
}
```

and not 

``` scss
.element + .element {
  …
}
```

The linter should probably allow the use of a parent reference in the case of `& + &`, so that we can have this:

``` scss
.element {
  & + & { … }
}
```
